### DISCUSSION TITLE
.env not loading correctly

### DISCUSSION THREAD

**Discussion by Tizooo** [vercel/next.js#70956] [Help] [answered]
### Summary

i have a .env file which i want to load into my nextjs app but sometimes it does not recognize the .env file and dometimes it does but i cant use the values in some files. for example i have a getGuilds.js where i need the DISCORD_BOT_TOKEN which i try getting with process.env.DISCORD_BOT_TOKEN but it comes out as undefined. in other files i also use env values which work fine. below are screenshots of the filestructure, how i require .env and how my getGuilds.js looks like + what the output is
![image](https://github.com/user-attachments/assets/31562995-8367-497c-b48f-0f7539912d9a)
![image](https://github.com/user-attachments/assets/8c0c1c1d-7f31-4df1-92f6-0da74f3447e8)

**Comment by Tizooo** [comment]
and my .env file looks like this: (note that i removed some values since they are private)

DISCORD_BOT_TOKEN=my_secret_here
DISCORD_CLIENT_SECRET=another_one
DISCORD_CLIENT_ID=849689558697377814
DISCORD_GUILD_ID=1244951898239275061
DISCORD_REDIRECT_URI=http://localhost:3000/api/auth/callback/discord

MONGODB_URI=SEECREEET

NEXTAUTH_URL=http://localhost:3000
NEXTAUTH_SECRET=SECRET

SESSION_SECRET=secret

PORT=3000

**Comment by Permanf** [comment]
Good to know: environment variables specified in this way will always be included in the JavaScript bundle, prefixing the environment variable name with NEXT_PUBLIC_ only has an effect when specifying them [through the environment or .env files](https://nextjs.org/docs/app/building-your-application/configuring/environment-variables).

Have you checked this out?

**Reply by Tizooo** [reply]
Yes, i know that if i add NEXT_PUBLIC_ to the variable i can also use it on the client side but since the getGuilds is not client side i should not need to use NEXT_PUBLIC_ here to be able to call varaibles? 

or am i missing something?

**Comment by wilsonadenuga** [comment]
Can you check the log on your console where you are running the server, not the browser? It should be visible there. The browser is client-side, so without prefixing the env variable with  NEXT_PUBLIC, it would always be undefined on the client.

**Reply by Tizooo** [reply]
i run the server in VScode. it doesent show the console.log there. i made a small sample setup which shows you the issue im facing. i left the DISCORD_CLIENT_SECRET in the .env file so the code works. you can check out my issue there. the getGuilds.js does not take ANY variables from the .env but the rest works fine.

to run it just use
- npm i
- npm run dev

[showcase.zip](https://github.com/user-attachments/files/17293366/showcase.zip)

**Comment by Tizooo** [comment, marked as answer]
i think i found the issue, i use "use client"; in the layout.js which makes everything clientside o-o

**Reply by wilsonadenuga** [reply]
[showcase_2.zip](https://github.com/user-attachments/files/17293945/showcase_2.zip)

fixed like so, you are correct about the layout.
